### PR TITLE
Add pickle and improve filter

### PR DESCRIPTION
**Pull request by jinwar**
adding small changes to the code

### CODE REVIEW

File: dascore/core/patch.py [hunk 1]
```diff
@@ -30,16 +30,19 @@ def __init__(
...
+                return
+            dims = dims if dims is not None else list(coords)
+            mixer = _AttrsCoordsMixer(attrs, coords, dims)
+            attrs, coords = mixer()
+            # get xarray coords from custom coords object
+            if isinstance(coords, Coords):
+                coords = coords._coords
+            self._data_array = DataArray(
+                data=data, dims=dims, coords=coords, attrs=attrs
+            )
```
Comment (d-chambers) [inline dascore/core/patch.py:45]: Hey @jinwar, so I added a few tests in tests/core/patch.py. I decided to create a new test case (class) since there will likely be several tests around empty patches (a common edge case).

It looks like this:

```python
class TestEmptyPatch:
    """Tests for empty patch objects."""

    def test_empty_patch_str(self):
        """An empty patch should have both a str and repr of nonzero length."""
        patch = Patch()
        assert len(str(patch))
        assert len(repr(patch))

    def test_has_attrs(self):
        """An empty test patch should have attrs with null(ish) values."""
        patch = Patch()
        attrs = patch.attrs
        assert len(attrs)
        # Check that default values exist but are null (of appropriate dtype)
        names = ['time', 'distance']
        fields = ['d_{x}', '{x}_min', "{x}_max"]
        for name in names:
            for field in fields:
                value = attrs[field.format(x=name)]
                assert pd.isnull(value)
```

File: dascore/core/patch.py [hunk 2]
```diff
@@ -139,12 +142,19 @@ def attrs(self) -> FrozenDict:
...
     detrend = dascore.proc.detrend
     pass_filter = dascore.proc.pass_filter
     aggregate = dascore.proc.aggregate
     abs = dascore.proc.abs
 
+    # --- io funcs
+
+    save_pickle = dascore.utils.save_pickle
+    load_pickle = dascore.utils.load_pickle
+
```
Comment (d-chambers) [inline dascore/core/patch.py:157]: We should create a new io module for pickle. However, I was just thinking about refactoring this so we can leave this until we refactor. 

Normally IO would look like this (in ObsPy)
```python
import dascore as dc
patch = dc.read(picke_path, format='pickle')  # format is optional, it can auto-detect
patch.write("output_path.pkl", "pickle")
```

File: dascore/core/patch.py (hunk 1 above)
Comment (d-chambers) [reply dascore/core/patch.py:45]: I had to refactor the `Patch.__init__` method slightly to get the tests to pass.

**Comment by d-chambers** [comment]
Hey @jinwar, sorry about this but there were a few file ending issues (differences between linux and windows) that mangled some of the files, which is why it shows so many files changed, but it should be good when we merge. 

I added some tests for the empty patch and started a io.pickle module. However, I just had a good idea for refactoring the IO to allow for easily adding parsers (as we discussed last week) so I am going to merge this for now then refactor a bit in a subsequent PR.

**Comment by d-chambers** [comment]
I moved the `isel` and `sel` methods to a new PR #14 as I think they merit some more consideration in light of the conversation in #11
